src/net.cpp

```diff
@@ -1571,8 +1579,53 @@ void StopMapPort()
 #endif
 
 
+void CConnman::DNSCachesUpdate() const
+{
+    // Only reachable nodes do this, because those are already easy to find.
+    // Unless the node is explicitly set to be non-reachable (-listen=0),
+    // the only way to check if it is reachable is to observe inbound connections.
+    // This check is made every time, because a node may become non-reachable
+    // due to external factors (e.g. local network settings update).
+    if (CountInboundConnections() == 0) return;
+
+    // Every time a new widely-requests service bit is deployed, this list should be
+    // updated to invalide records for that service bit.
+    std::vector<uint64_t> SERVICE_BITS_COMBINATIONS{
+        // Legacy nodes
+        NODE_NETWORK,
+        NODE_NETWORK_LIMITED,
+        // Segwit-supporting nodes
+        NODE_NETWORK | NODE_WITNESS,
+        NODE_NETWORK_LIMITED | NODE_WITNESS,
+        // Nodes supporting Bloom Filters per BIP 111
+        NODE_BLOOM,
+        NODE_NETWORK | NODE_BLOOM,
+        NODE_NETWORK_LIMITED | NODE_BLOOM,
+        NODE_NETWORK | NODE_WITNESS | NODE_BLOOM,
+        NODE_NETWORK_LIMITED | NODE_WITNESS | NODE_BLOOM};
+
+    for (const std::string& seed : Params().DNSSeeds()) {
+        if (interruptNet) {
+            return;
+        }
 
+        // first query as if client doesn't support service bits first
+        std::vector<CNetAddr> ips;
+        LookupHost(seed, ips, 1, true);
+        // TODO: Here and below, make sure downloaded IPs are not very different
+        // from what stored locally, to identify AddrMan or DNS poisoning.
 
+        // query for all widely used combinations of service bits
+        for (uint64_t service_bits_combination : SERVICE_BITS_COMBINATIONS) {
```

Comment:
Should these really be done all at the same time? No new node would batch them like this...?

Reply:
Well, one private node in the subnet will request one combination, another private node will request another one...

Why would you want the already public node separate cache updates in time?